src/spectral.ts

```diff
@@ -11,23 +13,94 @@ import {
   isOpenApiv3,
 } from '@stoplight/spectral';
 
-import { tryCatch } from 'fp-ts/lib/TaskEither';
-import { toError } from 'fp-ts/lib/Either';
-
-export const createSpectral = (ruleset: string) => {
-  return tryCatch(async () => {
-    const spectral = new Spectral({ resolver: httpAndFileResolver });
-    spectral.registerFormat('oas2', isOpenApiv2);
-    spectral.registerFormat('oas3', isOpenApiv3);
-    spectral.registerFormat('json-schema', isJSONSchema);
-    spectral.registerFormat('json-schema-loose', isJSONSchemaLoose);
-    spectral.registerFormat('json-schema-draft4', isJSONSchemaDraft4);
-    spectral.registerFormat('json-schema-draft6', isJSONSchemaDraft6);
-    spectral.registerFormat('json-schema-draft7', isJSONSchemaDraft7);
-    spectral.registerFormat('json-schema-2019-09', isJSONSchemaDraft2019_09);
-    await spectral.loadRuleset(ruleset);
-    return spectral;
-  }, toError);
+import * as IOEither from 'fp-ts/lib/IOEither';
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as E from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+
+import { info } from '@actions/core';
+import { pluralizer } from './utils';
+
+const evaluateNumberOfExceptions = (exceptions: Record<string, string[]>) => {
+  const reduced = Object.keys(exceptions).reduce(
+    (acc, cur) => {
+      acc.uniqueFilePaths.add(cur.split('#')[0]);
+      acc.numberOfExceptions += exceptions[cur].length;
+      return acc;
+    },
+    { numberOfExceptions: 0, uniqueFilePaths: new Set() }
+  );
+
+  return {
+    numberOfExceptions: reduced.numberOfExceptions,
+    numberOfFiles: reduced.uniqueFilePaths.size,
+  };
+};
+
+const normalizeRulesetPath = (rulesetPath: string): [string] | undefined => {
+  if (rulesetPath.length === 0) {
+    info(`Loading built-in rulesets...`);
+    return undefined;
+  }
+
+  info(`Loading ruleset '${rulesetPath}'...`);
+  return [rulesetPath];
 };
 
-export const runSpectral = (spectral: Spectral, parsed: string) => tryCatch(() => spectral.run(parsed), toError);
+const retrieveSpectralPackageVersion = (): IOEither.IOEither<Error, string> =>
+  IOEither.tryCatch<Error, string>(() => {
+    const x = require('../node_modules/@stoplight/spectral/package.json');
+    return String(x.version);
+  }, E.toError);
```

Comment:
You do not need to make a function of this
```suggestion
const retrieveSpectralPackageVersion =
  IOEither.tryCatch<Error, string>(() => {
    const x = require('../node_modules/@stoplight/spectral/package.json');
    return String(x.version);
  }, E.toError);
```

Reply:
@XVincentX Duh. I now realize I didn't take this into account. Sorry :-(